config/i18n/all-langs.ts

```diff
@@ -148,28 +148,69 @@ export const i18nextCodes = {
 };
 
 // These are for the language selector dropdown menu in the footer
-export const langDisplayNames = {
-  english: 'English',
-  espanol: 'Español',
-  chinese: '中文（简体字）',
-  'chinese-traditional': '中文（繁體字）',
-  italian: 'Italiano',
-  portuguese: 'Português',
-  ukrainian: 'Українська',
-  japanese: '日本語'
-};
+export enum LangNames {
+  English = 'English',
+  Espanol = 'Español',
+  Chinese = '中文（简体字）',
+  'Chinese-traditional' = '中文（繁體字）',
+  Italian = 'Italiano',
+  Portuguese = 'Português',
+  Ukrainian = 'Українська',
+  Japanese = '日本語'
+}
 
 /* These are for formatting dates and numbers. Used with JS .toLocaleString().
  * There's an example in profile/components/Camper.js
  * List: https://github.com/unicode-cldr/cldr-dates-modern/tree/master/main
  */
-export const langCodes = {
-  english: 'en-US',
-  espanol: 'es-419',
-  chinese: 'zh',
-  'chinese-traditional': 'zh-Hant',
-  italian: 'it',
-  portuguese: 'pt-BR',
-  ukrainian: 'uk',
-  japanese: 'ja'
-};
+export enum LangCodes {
+  English = 'en-US',
+  Espanol = 'es-419',
+  Chinese = 'zh',
+  'Chinese-traditional' = 'zh-Hant',
+  Italian = 'it',
+  Portuguese = 'pt-BR',
+  Ukrainian = 'uk',
+  Japanese = 'ja'
+}
+
+// returns a key-value from an enum for client locale which is initiated from a JSON file.
+// JSON does not support enums.
+
+export function langCodeIncludes(locale: string) {
+  const langCodeObj = Object.entries(LangCodes).map(([key, value]) => ({
+    id: key.toLocaleLowerCase(),
+    value: value
+  }));
+
+  const entryObj = (obj: Record<string, string>): boolean => {
+    if (obj.id == locale) {
+      return true;
+    }
+    return false;
+  };
+
+  const result = langCodeObj.filter(entryObj);
+
+  return result.length > 0 ? result[0]['value'] : LangCodes.English;
+}
```

Comment:
So, if we can't find the LanguageCode we should just throw an error, since we don't know how to deal with it.

Also, you can use [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to narrow the type of `locale` from `string` to an enum key.

```
function isPropertyOf<T>(obj: T, key: PropertyKey): key is keyof T {
  return Object.prototype.hasOwnProperty.call(obj, key);
}
```
should do the trick for both this and the function below.

Finally, `getLangCode` and `getLangName` are slightly more descriptive names.

Reply:
that seems fairly new!